src/TurbulenceClosures/vertically_implicit_diffusion_solver.jl

```diff
@@ -40,46 +40,49 @@ implicit_diffusion_solver(::ExplicitTimeDiscretization, args...; kwargs...) = no
 
 @inline function ivd_upper_diagonalᵃᵃᶜ(i, j, k, grid, clock, Δt, κ⁻⁻ᶠ, κ)
     κᵏ⁺¹ = κ⁻⁻ᶠ(i, j, k+1, grid, clock, κ)
-    return - Δt * κ_Δz²(i, j, k, k+1, grid, κᵏ⁺¹)
+
+    return ifelse(k < 1,
+                  zero(eltype(grid)),
+                  - Δt * κ_Δz²(i, j, k, k+1, grid, κᵏ⁺¹))
```

Comment:
I think the special cases (k=1 & k=Nz) should be switched, i.e., we should take care of k = 0 in lower-diag function and take care of k=Nz in upper-diag function.

Reply:
since there's no test I think we are free to switch. @jm-c do you want me to do it or do you want to do it?

Reply:
I am currently checking other index related things here and will try to run it after that. From there I could push changes to his PR.